### PR TITLE
Migrate host network detection to use workloadmeta

### PR DESCRIPTION
**Pull request by juliogreff**
### What does this PR do?

We're trying to clean up some old code from the kubeUtil, so anything
that fetches pod information needs to be migrated to workloadmeta. This
commit refactors host network detection to use workloadmeta instead of
pulling data directly from the kubelet. Since this happens at startup,
we also had to introduce a workloadmeta.Store.WaitForCollectors for
synchronization.

### Motivation

Cleaning up unnecessary calls to the kubelet now that workloadmeta is a viable (and better!) replacement.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/workloadmeta/types.go [hunk 1]
```diff
@@ -75,6 +75,10 @@ type Store interface {
 
 	// Dump lists the content of the store, for debugging purposes.
 	Dump(verbose bool) WorkloadDumpResponse
+
+	// WaitForCollectors waits until all collectors either start collecting
+	// data, are removed due to being disabled, or ctx is cancelled.
```
Comment (djmitche) [inline pkg/workloadmeta/types.go:80]: I would think you would want to wait until the collectors have completed their first collection - not just until they've started?

File: pkg/workloadmeta/types.go [hunk 2]
```diff
@@ -432,6 +436,7 @@ type KubernetesPod struct {
 	QOSClass                   string
 	KubeServices               []string
 	NamespaceLabels            map[string]string
+	HostNetwork                bool
```
Comment (djmitche) [inline pkg/workloadmeta/types.go:439]: These fields could probably all use a brief bit of doc -- this one at least is not clear to me from its name.

**Review by djmitche — COMMENTED** [review]
Looks good but has some build issues the GitHub actions have flagged

File: pkg/workloadmeta/types.go (hunk 1 above)
Comment (juliogreff) [reply pkg/workloadmeta/types.go:80]: True. That's effectively the same for the kubelet collector (it pulls on `Start`), so I'll amend the doc to document that expectation (and fix non-compliant collectors).

**Review by paulcacheux — APPROVED** [review]

**Pull request closed without merging**